ksql-engine/src/main/java/io/confluent/ksql/metrics/MetricCollector.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package io.confluent.ksql.metrics;
+
+import io.confluent.common.metrics.Metrics;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+
+import java.util.Date;
+import java.util.Map;
+
+public class MetricCollector extends AbstractMetricCollector {
+  private ConsumerCollector consumerCollector;
+  private ProducerCollector producerCollector;
+  private final Date created = new Date();
+  private String clientId;
+
+  /**
+   * Created by via registration as an Interceptor (hence the ugly binding to the MetricCollector)
+   */
+  public MetricCollector(){
+    Metrics metrics = MetricCollectors.addCollector(this);
+
+    // Note: I will probably break these apart
+    consumerCollector = new ConsumerCollector(metrics);
```

Comment:
As noted above - i think if we move these into `configure` we can also pass in the id at construction time